src/schedule/ucc_schedule.h

```diff
@@ -23,13 +25,26 @@ typedef enum {
 } ucc_event_t;
 
 typedef struct ucc_coll_task ucc_coll_task_t;
+
+typedef struct ucc_schedule ucc_schedule_t;
+
 typedef struct ucc_base_team ucc_base_team_t;
-typedef ucc_status_t (*ucc_task_event_handler_p)(ucc_coll_task_t *parent,
-                                                 ucc_coll_task_t *task);
+
 typedef ucc_status_t (*ucc_coll_post_fn_t)(ucc_coll_task_t *task);
-typedef ucc_status_t (*ucc_coll_triggered_post_fn_t)(ucc_ee_h ee, ucc_ev_t *ev, ucc_coll_task_t *task);
+
+typedef ucc_status_t (*ucc_coll_progress_fn_t)(ucc_coll_task_t *task);
+
 typedef ucc_status_t (*ucc_coll_finalize_fn_t)(ucc_coll_task_t *task);
 
+typedef ucc_status_t (*ucc_task_event_handler_p)(ucc_coll_task_t *parent,
+                                                 ucc_coll_task_t *task);
+
+typedef ucc_status_t (*ucc_coll_triggered_post_setup_fn_t)(ucc_coll_task_t *task);
```

Comment:
lets also leave a short comment of what this fn should do:
smth like called during trigger event handler before starting the executor.